### ISSUE TITLE
Identified unknown Google Workspace group logic functionality with multi-party approval

### ISSUE THREAD

**Issue by GangGreenTemperTatum** [GoogleCloudPlatform/jit-access#82]
Hello World! 👋🏼 

## **Background**:
- Demo environment (users are already associated to numerous groups within Google Workspace)
- Those same Google Groups (some of) are currently set (inherited) as GCP IAM principles with permissions across org/projects
  - These IAM principles have different GCP permissions across different projects
  - A flavor or org-inherited and per-project
- Successfully implemented JIT using Cloud Function as well as multi-party approval
- We have successfully tested the process of MPA from start to finish with **some** users and works successfully
  - I have even gone to the extent of testing the access and verifying the constraint condition present in GCP IAM UI

<br>
Furthermore:
<br>

- We have tried to narrow down this issue by (via code) allowing one Google Workspace group, a set of defined roles for all GCP projects which works for some users and not others

## **Problem**:

- Depending on the specific scenario, I.E user and associated Google Workspace group, only select roles are available per-project

## **Hypothesis**:

- To prove this, I performed a strategic approach to applying per single-group membership to a user account and continuously testing the same request in JIT app UI. From testing, it follows suit of process of elimination of the Google Workspace groups.
- **TL;DR**.. It seems that based on some logic, different groups the user is associated to on Workspace, is reflected within the JIT App UI (dependent on the project and roles available) are being chosen almost at random
 -  It does not seem to be order from what is displayed within Google Workspace UI (tested ✅ )
 -  It does not seem to be A-Z/top-down (tested ✅ )

## **Potentially Related**:

- #81 

## **Request**:

- Is anyone able to elaborate on what this logic is? IIAMC here
- The preferrable would be a logic change that can be defined from the JIT application configuration

<br>

PLMK if you need anything else (it's a difficult one to explain on paper) and TYIA! 🙏🏼

**Comment by jpassing** [comment]
If I understand correctly, then JIT Access only shows a (seemingly arbitrary) subset of the roles that you'd expect to see. Is that right?

The Asset Inventory API (which JIT Access uses behind the scenes to find role bindings) does impose some limits on how many resources and roles it can analyze at once. If JIT Access hits one of these limits, you should see a warning pop up at the bottom of the screen. But I suppose that wasn't the case?

**Comment by jpassing** [comment]
When you deployed JIT Access, did you set `RESOURCE_SCOPE`  to a project, folder, or the entire organization? And can you confirm that the role bindings in question (the ones for `group-email@domain` and `group2-email@domain`) are within that scope? If, say, you set `RESOURCE_SCOPE` to a folder and then create a role binding in the organization, then JIT Access will ignore that binding as it's not in scope.

To analzye the issue further, it might be helpful to compare the JIT Access results to what you see in Policy Analzyer:

1. Open [Policy Analyyer](https://console.cloud.google.com/iam-admin/analyzer/query)
2. Set **Scope** to match your `RESOURCE_SCOPE`  configuration in JIT Access
3. Set **Resource** to `//cloudresourcemanager.googleapis.com/projects/PROJECT_ID`  were `PROJECT_ID` is the project that you selected in JIT Access
4. Set **Principal** to the user you signed in as in JIT Access (who is a member of  `group-email@domain` and `group2-email@domain`)
5. Click **Analyze > Run query**

Note that the result includes both eliible role bindings  (i.e., bindings with IAM condition `has({}.jitAccessConstraint)` or `has({}.jitAccessConstraint)`) as well as permanent bindings. 

Does the result include all the roles that you'd expect to see (that is, all roles assigned to `group-email@domain` and `group2-email@domain`)?  Or are there items missing here too?
